src/drivers/net/e1000.c

```diff
@@ -197,16 +201,18 @@ static void e1000_rx(struct net_device *dev) {
 			skb = rx_skbs[cur];
 			assert (skb);
 
-			rx_skbs[cur]= skb_alloc(ETH_FRAME_LEN);
+			rx_skbs[cur] = skb_alloc(E1000_MAX_RX_LEN);
 			if (rx_skbs[cur] == NULL)
 				panic("Can't allocate rx_skbs %d", cur);
 			rx_descs[cur].buffer_address = (uint32_t) rx_skbs[cur]->mac.raw;
 
-
+			len = rx_descs[cur].length - E1000_RX_CHECKSUM_LEN;
+			skb = skb_realloc(len, skb);
+			if (!skb) {
+				panic("Can't reallocate rx_skb");
```

Comment:
That's weird anyway. I.e. instead of dropping the packet, we just drop the system? :)